gson/src/main/java/com/google/gson/internal/reflect/ReflectionHelper.java

```diff
@@ -24,65 +24,81 @@ public class ReflectionHelper {
 
   private ReflectionHelper() {}
 
-  /**
-   * Tries making the field accessible, wrapping any thrown exception in a {@link JsonIOException}
-   * with descriptive message.
-   *
-   * @param field field to make accessible
-   * @throws JsonIOException if making the field accessible fails
-   */
-  public static void makeAccessible(Field field) throws JsonIOException {
-    makeAccessible("field '" + field.getDeclaringClass().getName() + "#" + field.getName() + "'", field);
-  }
-
-  /**
-   * Tries making the constructor accessible, wrapping any thrown exception in a {@link JsonIOException}
-   * with descriptive message.
-   *
-   * @param constructor constructor to make accessible
-   * @throws JsonIOException if making the constructor accessible fails
-   */
-  public static void makeAccessible(Constructor<?> constructor) throws JsonIOException {
-    makeAccessible(
-            "constructor " + constructor + " in " + constructor.getDeclaringClass().getName(),
-            constructor
-    );
-  }
-
   /**
    * Internal implementation of making an {@link AccessibleObject} accessible.
    *
-   * @param description describe what we are attempting to make accessible
    * @param object the object that {@link AccessibleObject#setAccessible(boolean)} should be called on.
    * @throws JsonIOException if making the object accessible fails
    */
-  private static void makeAccessible(String description, AccessibleObject object) throws JsonIOException {
+  public static void makeAccessible(AccessibleObject object) throws JsonIOException {
     try {
       object.setAccessible(true);
     } catch (Exception exception) {
-      throw new JsonIOException("Failed making " + description + "' accessible; either change its visibility "
-              + "or write a custom TypeAdapter for its declaring type", exception);
+      String description = getAccessibleObjectDescription(object, false);
+      throw new JsonIOException("Failed making " + description + " accessible; either increase its visibility "
+              + "or write a custom TypeAdapter for its declaring type.", exception);
     }
   }
 
+  /**
+   * Returns a short string describing the {@link AccessibleObject} in a human-readable way.
+   *
+   * @param object object to describe
+   * @param uppercaseFirstLetter whether the first letter of the description should be uppercased
+   */
+  public static String getAccessibleObjectDescription(AccessibleObject object, boolean uppercaseFirstLetter) {
+    String description;
+
+    if (object instanceof Field) {
+      Field field = (Field) object;
+      description = "field '" + field.getDeclaringClass().getName() + "#" + field.getName() + "'";
+    } else if (object instanceof Method) {
+      Method method = (Method) object;
+
+      StringBuilder methodSignatureBuilder = new StringBuilder(method.getName());
```

Comment:
This all seems like a lot of trouble to go to. It seems to me that the default `toString()` of `Method` and `Constructor` is good enough, isn't it?

Reply:
The reason why I added this is that `toString()` can be a bit verbose because it:
- includes all modifiers (`public`, `final`, `transient`, ...)
- uses qualified names for parameter types
- for fields also includes the field type

This implementation here omits modifiers and uses simple names. Though on the other hand maybe there are situations where at least some modifiers are relevant?

Reply:
OK, I still think `toString()` would be good enough, but I agree with you about the noise it can introduce.